pkg/api/handlers/compat/containers_stop.go

```diff
@@ -27,31 +34,25 @@ func StopContainer(w http.ResponseWriter, r *http.Request) {
 	}
 
 	name := utils.GetName(r)
-	con, err := runtime.LookupContainer(name)
-	if err != nil {
-		utils.ContainerNotFound(w, name, err)
-		return
-	}
 
-	state, err := con.State()
-	if err != nil {
-		utils.InternalServerError(w, errors.Wrapf(err, "unable to get state for Container %s", name))
-		return
+	options := entities.StopOptions{
+		All:     query.All,
+		Ignore:  query.Ignore,
+		Timeout: &query.Timeout,
 	}
-	// If the Container is stopped already, send a 304
-	if state == define.ContainerStateStopped || state == define.ContainerStateExited {
```

Comment:
Need to retain this check so we can return `StatusNotModified`

Reply:
Yuck, Fixed.